app/scripts/models/account.js

```diff
@@ -323,7 +328,15 @@ define(function (require, exports, module) {
       .then(function (updatedSessionData) {
         self.set(updatedSessionData);
 
-        if (! self.get('verified')) {
+        // `recoveryEmailStatus` does not return a reason why the
+        // sessionToken is not valid, only that it is invalid.
```

Comment:
@shane-tomlinson This was one of the PR comments from auth-server. The `/recovery_email/status` endpoint now returns additional information `emailVerified` and `sessionVerified`, ref https://github.com/mozilla/fxa-auth-server/blob/verify-token-full/docs/api.md#response-13.